xarray/tests/test_backends.py

```diff
@@ -1535,6 +1535,14 @@ def test_keep_chunksizes_if_no_original_shape(self) -> None:
                 ds["x"].encoding["chunksizes"], actual["x"].encoding["chunksizes"]
             )
 
+    def test_preferred_chunks_is_present(self) -> None:
+        ds = Dataset({"x": [1, 2, 3]})
+        chunksizes = (2,)
+        ds.variables["x"].encoding = {"chunksizes": chunksizes}
+
+        with self.roundtrip(ds) as actual:
+            assert actual["x"].encoding["preferred_chunks"] == {"x": 2}
+
```

Comment:
This test is a good start. However, I would like to see a more thorough set of tests that set up NetCDF files with various chunk schemas and confirm that the `auto`/`{}` options return expected outputs after loading.

Reply:
Ok, I'll do that. Is this file the right place for these tests?

Reply:
@jhamman done, tell me what you think :)